src/utilities/oas-validator.ts

```diff
@@ -32,48 +32,52 @@ class OasValidator {
 
   validateParameters = async (
     operationId: string,
-    parameters: WrappedParameterExamples,
-  ): Promise<void> => {
-    const unwrappedParameters = ParameterWrapper.unwrapParameters(parameters);
+    parameters: ParameterExamples,
+  ): Promise<ValidationFailure[]> => {
     const parameterSchema: {
       [parameterName: string]: Parameter;
     } = {};
     const operation = await this.schema.getOperation(operationId);
     if (!operation) {
-      throw new InvalidOperationIdError(operationId);
+      return [new InvalidOperationId(operationId)];
     }
     const requiredParameters = operation.parameters
       .filter((parameter) => parameter.required)
       .map((parameter) => parameter.name);
 
-    const presentParameterNames = Object.keys(unwrappedParameters);
+    const presentParameterNames = Object.keys(parameters);
     const missingRequiredParameters = requiredParameters?.filter(
       (parameterName) => !presentParameterNames.includes(parameterName),
     );
 
     if (missingRequiredParameters.length > 0) {
-      throw new MissingRequiredParametersError(missingRequiredParameters);
+      return [new MissingRequiredParameters(missingRequiredParameters)];
```

Comment:
Should we consider validating present parameters even if some required ones are missing?

Reply:
yeah that's a good thought. I'll update this.